azure-iot-device/azure/iot/device/common/async_adapter.py

```diff
@@ -33,20 +36,46 @@ class AwaitableCallback(object):
     """A sync callback whose completion can be waited upon.
     """
 
-    def __init__(self, callback):
-        """Creates an instance of an AwaitableCallback from a callback function.
-
-        :param callback: Callback function to be made awaitable.
+    def __init__(self, return_arg_name=None):
```

Comment:
I agree, but once we added an `error=` mechanism, this ceased to be general purpose and then became a mechanism for returning results (or exceptions).  Since all we need _right now_ is a mechanism to return results or errors, this seemed like a good tradeoff to make.  

---
In reply to: [318828869](https://github.com/Azure/azure-iot-sdk-python-preview/pull/174#discussion_r318828869) [](ancestors = 318828869)